### PR TITLE
Build and run CommCareHQ in a docker container

### PR DESCRIPTION
**Pull request by charlesfleche**
PURPOSE
Allowing non-tech people to install and run offline demo instances of CommCareHQ for training / remote app building purpose.

NOTE
This PR adds the necessary files to create and run a CommCareHQ [docker](https://en.wikipedia.org/wiki/Docker_%28software%29) image, allowing to easily run local instances on local servers / laptops. It also updates the install.sh script.

TRY IT
- [install docker](http://docs.docker.com/installation)
- run {commcare-hq}/docker/docker-start.sh
- connect to http://localhost:8000 (or your boot2docker instance address on OSX / Windows)
- login: example@example.com / password: example

TODO
This PR is probably not ready for integration quite yet. But as I won't be online until next week it gives me the opportunity to show what I'm working on these days, collect feature requests and allow to build local docker images as it is going to take a while to push my layers from Myanmar… These tasks are my next moves :
- try this on OSX / Windows
- write a user documentation on the CommCare wiki
- allow to stop / restart the containers
- allow app building
- add elasticsearch

CHANGES
- Dockerfile creates a Dedian Wheezy based CommCareHQ image

Wheezy has been choosen as a base image because the official redis, postgres and couchdb are based on the same image, potentialy reducing the amount of data to download.

The apt-get calls should be concatenated into a single one. My bandwidth here is just too thin to allow me to recreate entire install layers while building an image.
- docker/docker-start.sh instanciates the databases and commcarehq

Databases initializations (users, db, django) is done by instanciating temporary couchdb / postgres / commcarehq images that already contain the necessary binaries (like psql).
- docker/docker-rm.sh deletes each db + commcarehq instances
- install.sh
  - Automatically download the jdk if required
  - Update the Ubuntu package name for add-apt-repository
  - Removed now unused dependencies to couchdb-lucene and egenix-mx-base

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 67 errors:

* docker/localsettings.py, line 2 - E265 block comment should start with '# '
* docker/localsettings.py, line 2 - W291 trailing whitespace
* docker/localsettings.py, line 15 - E265 block comment should start with '# '
* docker/localsettings.py, line 16 - E501 line too long (127 > 115 characters)
* docker/localsettings.py, line 18 - E265 block comment should start with '# '
* docker/localsettings.py, line 19 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E262 inline comment should start with '# '
* docker/localsettings.py, line 25 - E265 block comment should start with '# '
* docker/localsettings.py, line 27 - E265 block comment should start with '# '
* docker/localsettings.py, line 30 - E265 block comment should start with '# '
* docker/localsettings.py, line 33 - E265 block comment should start with '# '
* docker/localsettings.py, line 48 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 48 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 52 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E501 line too long (173 > 115 characters)
* docker/localsettings.py, line 54 - E265 block comment should start with '# '
* docker/localsettings.py, line 65 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 68 - E265 block comment should start with '# '
* docker/localsettings.py, line 72 - E265 block comment should start with '# '
* docker/localsettings.py, line 78 - E265 block comment should start with '# '
* docker/localsettings.py, line 89 - E265 block comment should start with '# '
* docker/localsettings.py, line 99 - E202 whitespace before '}'
* docker/localsettings.py, line 101 - E265 block comment should start with '# '
* docker/localsettings.py, line 103 - W291 trailing whitespace
* docker/localsettings.py, line 104 - W291 trailing whitespace
* docker/localsettings.py, line 106 - W291 trailing whitespace
* docker/localsettings.py, line 111 - E265 block comment should start with '# '
* docker/localsettings.py, line 121 - E265 block comment should start with '# '
* docker/localsettings.py, line 160 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 161 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 162 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E501 line too long (137 > 115 characters)
* docker/localsettings.py, line 164 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - W291 trailing whitespace
* docker/localsettings.py, line 199 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 200 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 208 - E265 block comment should start with '# '
* docker/localsettings.py, line 211 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 211 - E113 unexpected indentation
* docker/localsettings.py, line 212 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 212 - E113 unexpected indentation
* docker/localsettings.py, line 213 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 213 - E113 unexpected indentation
* docker/localsettings.py, line 216 - E265 block comment should start with '# '
* docker/localsettings.py, line 218 - E225 missing whitespace around operator
* docker/localsettings.py, line 219 - E225 missing whitespace around operator
* docker/localsettings.py, line 223 - E265 block comment should start with '# '
* docker/localsettings.py, line 228 - W293 blank line contains whitespace
* docker/localsettings.py, line 231 - W293 blank line contains whitespace
* docker/localsettings.py, line 279 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 280 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 283 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 284 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 294 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 295 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 296 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 298 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 299 - E501 line too long (116 > 115 characters)
* docker/localsettings.py, line 301 - E501 line too long (121 > 115 characters)
* docker/localsettings.py, line 302 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 303 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 307 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 308 - E501 line too long (118 > 115 characters)

**Comment by charlesfleche** [comment]
TODO
* Allow app building

File: requirements/requirements.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@ django-kombu==0.9.1
 django-rosetta==0.7.1
 django-tastypie==0.9.14
 djtables==0.1.1
-egenix-mx-base==3.2.8
+#egenix-mx-base==3.2.8
```
Comment (czue) [inline requirements/requirements.txt]: was this intentional?

**Comment by czue** [comment]
@charlesfleche this looks really cool!

fyi, you may want to take a look at a similar project we're working on with ansible that we're using to deploy to our production infrastructure: https://github.com/dimagi/commcarehq-ansible (I realize this is a bit late to possibly be super useful to you).

I'm also totally unqualified to review this PR. We may have to wait for @dannyroberts (who is on vacation) or @TylerSheffels . I'm also applying the "don't merge" tag based on your comment that it's not ready, but let us know when you think it is.

File: docker/localsettings.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+####### Database config. This assumes Postgres ####### 
+
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql_psycopg2',
+        'NAME': 'commcarehq',
+        'USER': 'commcarehq',
+        'PASSWORD': 'commcarehq',
+        'HOST': 'postgres',
+        'PORT': '5432'
```
Comment (snopoke) [inline docker/localsettings.py]: host and port should come from the `POSTGRES_PORT_5432_TCP_ADDR` and `POSTGRES_PORT_5432_TCP_PORT` environment variables

Comment (snopoke) [inline docker/localsettings.py]: Shouldn't `port` use the `POSTGRES_PORT_5432_TCP_PORT` environment variable?

File: docker/localsettings.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+# prod settings
+SOIL_DEFAULT_CACHE = "redis"
+SOIL_BACKEND = "soil.CachedDownload"
+
+# reports cache
+REPORT_CACHE = 'default'  # or e.g. 'redis'
+
+redis_cache = {
+    'BACKEND': 'redis_cache.cache.RedisCache',
+    'LOCATION': 'redis:6379:0',
```
Comment (snopoke) [inline docker/localsettings.py]: Also should probably use `REDIS_PORT_6379_TCP_PORT` env var for port number.

**Comment by snopoke** [comment]
Super cool @charlesfleche! I've been wanting to do something like this for ages. Re Cory's comment about ansible, I found this interesting article: http://www.ansible.com/docker

**Comment by snopoke** [comment]
Need to run `sudo docker build -t charlesfleche/commcarehq .` before running `docker-start.sh`

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ && apt-get -y install python-pip \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN pip install --upgrade pip
+
+COPY requirements/requirements.txt /tmp/requirements.txt
+
+RUN apt-get update \
+ && apt-get -y install \
+      mysql-client \
```
Comment (snopoke) [inline Dockerfile]: don't need mysql client

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      postgresql-client \
+      sqlite3 \
```
Comment (snopoke) [inline Dockerfile]: probably don't need that sqlite3 either

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+RUN apt-get update \
+ && apt-get -y install \
+      python-dev \
+      --no-install-recommends
+
+RUN pip install -r /tmp/requirements.txt
+
+WORKDIR /usr/src/commcare-hq
+
+COPY . /usr/src/commcare-hq
```
Comment (snopoke) [inline Dockerfile]: Could make this a volume that gets mapped to the source folder on the host machine.

**Comment by dimagimon** [comment]
There are 67 errors:

* docker/localsettings.py, line 2 - E265 block comment should start with '# '
* docker/localsettings.py, line 2 - W291 trailing whitespace
* docker/localsettings.py, line 15 - E265 block comment should start with '# '
* docker/localsettings.py, line 16 - E501 line too long (127 > 115 characters)
* docker/localsettings.py, line 18 - E265 block comment should start with '# '
* docker/localsettings.py, line 19 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E262 inline comment should start with '# '
* docker/localsettings.py, line 25 - E265 block comment should start with '# '
* docker/localsettings.py, line 27 - E265 block comment should start with '# '
* docker/localsettings.py, line 30 - E265 block comment should start with '# '
* docker/localsettings.py, line 33 - E265 block comment should start with '# '
* docker/localsettings.py, line 48 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 48 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 52 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E501 line too long (173 > 115 characters)
* docker/localsettings.py, line 54 - E265 block comment should start with '# '
* docker/localsettings.py, line 65 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 68 - E265 block comment should start with '# '
* docker/localsettings.py, line 72 - E265 block comment should start with '# '
* docker/localsettings.py, line 78 - E265 block comment should start with '# '
* docker/localsettings.py, line 89 - E265 block comment should start with '# '
* docker/localsettings.py, line 99 - E202 whitespace before '}'
* docker/localsettings.py, line 101 - E265 block comment should start with '# '
* docker/localsettings.py, line 103 - W291 trailing whitespace
* docker/localsettings.py, line 104 - W291 trailing whitespace
* docker/localsettings.py, line 106 - W291 trailing whitespace
* docker/localsettings.py, line 111 - E265 block comment should start with '# '
* docker/localsettings.py, line 121 - E265 block comment should start with '# '
* docker/localsettings.py, line 160 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 161 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 162 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E501 line too long (137 > 115 characters)
* docker/localsettings.py, line 164 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - W291 trailing whitespace
* docker/localsettings.py, line 199 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 200 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 208 - E265 block comment should start with '# '
* docker/localsettings.py, line 211 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 211 - E113 unexpected indentation
* docker/localsettings.py, line 212 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 212 - E113 unexpected indentation
* docker/localsettings.py, line 213 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 213 - E113 unexpected indentation
* docker/localsettings.py, line 216 - E265 block comment should start with '# '
* docker/localsettings.py, line 218 - E225 missing whitespace around operator
* docker/localsettings.py, line 219 - E225 missing whitespace around operator
* docker/localsettings.py, line 223 - E265 block comment should start with '# '
* docker/localsettings.py, line 228 - W293 blank line contains whitespace
* docker/localsettings.py, line 231 - W293 blank line contains whitespace
* docker/localsettings.py, line 279 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 280 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 283 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 284 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 294 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 295 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 296 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 298 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 299 - E501 line too long (116 > 115 characters)
* docker/localsettings.py, line 301 - E501 line too long (121 > 115 characters)
* docker/localsettings.py, line 302 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 303 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 307 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 308 - E501 line too long (118 > 115 characters)

**Comment by dimagimon** [comment]
There are 67 errors:

* docker/localsettings.py, line 2 - E265 block comment should start with '# '
* docker/localsettings.py, line 2 - W291 trailing whitespace
* docker/localsettings.py, line 15 - E265 block comment should start with '# '
* docker/localsettings.py, line 16 - E501 line too long (127 > 115 characters)
* docker/localsettings.py, line 18 - E265 block comment should start with '# '
* docker/localsettings.py, line 19 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E262 inline comment should start with '# '
* docker/localsettings.py, line 25 - E265 block comment should start with '# '
* docker/localsettings.py, line 27 - E265 block comment should start with '# '
* docker/localsettings.py, line 30 - E265 block comment should start with '# '
* docker/localsettings.py, line 33 - E265 block comment should start with '# '
* docker/localsettings.py, line 48 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 48 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 52 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E501 line too long (173 > 115 characters)
* docker/localsettings.py, line 54 - E265 block comment should start with '# '
* docker/localsettings.py, line 65 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 68 - E265 block comment should start with '# '
* docker/localsettings.py, line 72 - E265 block comment should start with '# '
* docker/localsettings.py, line 78 - E265 block comment should start with '# '
* docker/localsettings.py, line 89 - E265 block comment should start with '# '
* docker/localsettings.py, line 99 - E202 whitespace before '}'
* docker/localsettings.py, line 101 - E265 block comment should start with '# '
* docker/localsettings.py, line 103 - W291 trailing whitespace
* docker/localsettings.py, line 104 - W291 trailing whitespace
* docker/localsettings.py, line 106 - W291 trailing whitespace
* docker/localsettings.py, line 111 - E265 block comment should start with '# '
* docker/localsettings.py, line 121 - E265 block comment should start with '# '
* docker/localsettings.py, line 160 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 161 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 162 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E501 line too long (137 > 115 characters)
* docker/localsettings.py, line 164 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - W291 trailing whitespace
* docker/localsettings.py, line 199 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 200 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 208 - E265 block comment should start with '# '
* docker/localsettings.py, line 211 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 211 - E113 unexpected indentation
* docker/localsettings.py, line 212 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 212 - E113 unexpected indentation
* docker/localsettings.py, line 213 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 213 - E113 unexpected indentation
* docker/localsettings.py, line 216 - E265 block comment should start with '# '
* docker/localsettings.py, line 218 - E225 missing whitespace around operator
* docker/localsettings.py, line 219 - E225 missing whitespace around operator
* docker/localsettings.py, line 223 - E265 block comment should start with '# '
* docker/localsettings.py, line 228 - W293 blank line contains whitespace
* docker/localsettings.py, line 231 - W293 blank line contains whitespace
* docker/localsettings.py, line 279 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 280 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 283 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 284 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 294 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 295 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 296 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 298 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 299 - E501 line too long (116 > 115 characters)
* docker/localsettings.py, line 301 - E501 line too long (121 > 115 characters)
* docker/localsettings.py, line 302 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 303 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 307 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 308 - E501 line too long (118 > 115 characters)

**Comment by charlesfleche** [comment]
Is it possible to ask digimon to ignore docker/localsettings.py ? I've greped the repo for localsettings.example.py and doesn't appear in any ignore list. I assume digimon has its own configuration outside of the commcarehq code base ?

File: requirements/requirements.txt (hunk 1 above)
Comment (charlesfleche) [inline requirements/requirements.txt]: Yes. I've greped the repo in search of these modules but could'nt find any import :
* mxDateTime
* mxTextTools
* mxStack
* mxTools
* mxProxy
* mxBeeBase

`for i in mxDateTime mxTextTools mxStack mxTools mxProxy mxBeeBase ; do grep -Ri $i . ; done`

File: Dockerfile (hunk 4 above)
Comment (charlesfleche) [inline Dockerfile]: Fixed in 1d74d65

File: Dockerfile (hunk 5 above)
Comment (charlesfleche) [inline Dockerfile]: Fixed in 1d74d65

File: Dockerfile (hunk 6 above)
Comment (charlesfleche) [inline Dockerfile]: Totally. Right now I'm focusing on deploying a ready to use image for trainings / offline installs, but next time I modify CommCare codes I'll take a look at mounting a clone in a container.

File: docker/localsettings.py (hunk 2 above)
Comment (charlesfleche) [inline docker/localsettings.py]: Fixed in 782cbb2

File: docker/localsettings.py (hunk 3 above)
Comment (charlesfleche) [inline docker/localsettings.py]: Fixed in 782cbb2

**Comment by dimagimon** [comment]
There are 67 errors:

* docker/localsettings.py, line 2 - E265 block comment should start with '# '
* docker/localsettings.py, line 2 - W291 trailing whitespace
* docker/localsettings.py, line 15 - E265 block comment should start with '# '
* docker/localsettings.py, line 16 - E501 line too long (127 > 115 characters)
* docker/localsettings.py, line 18 - E265 block comment should start with '# '
* docker/localsettings.py, line 19 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E262 inline comment should start with '# '
* docker/localsettings.py, line 25 - E265 block comment should start with '# '
* docker/localsettings.py, line 27 - E265 block comment should start with '# '
* docker/localsettings.py, line 30 - E265 block comment should start with '# '
* docker/localsettings.py, line 33 - E265 block comment should start with '# '
* docker/localsettings.py, line 48 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 48 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 52 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E501 line too long (173 > 115 characters)
* docker/localsettings.py, line 54 - E265 block comment should start with '# '
* docker/localsettings.py, line 65 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 68 - E265 block comment should start with '# '
* docker/localsettings.py, line 72 - E265 block comment should start with '# '
* docker/localsettings.py, line 78 - E265 block comment should start with '# '
* docker/localsettings.py, line 89 - E265 block comment should start with '# '
* docker/localsettings.py, line 99 - E202 whitespace before '}'
* docker/localsettings.py, line 101 - E265 block comment should start with '# '
* docker/localsettings.py, line 103 - W291 trailing whitespace
* docker/localsettings.py, line 104 - W291 trailing whitespace
* docker/localsettings.py, line 106 - W291 trailing whitespace
* docker/localsettings.py, line 111 - E265 block comment should start with '# '
* docker/localsettings.py, line 121 - E265 block comment should start with '# '
* docker/localsettings.py, line 160 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 161 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 162 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E501 line too long (137 > 115 characters)
* docker/localsettings.py, line 164 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - W291 trailing whitespace
* docker/localsettings.py, line 199 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 200 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 208 - E265 block comment should start with '# '
* docker/localsettings.py, line 211 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 211 - E113 unexpected indentation
* docker/localsettings.py, line 212 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 212 - E113 unexpected indentation
* docker/localsettings.py, line 213 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 213 - E113 unexpected indentation
* docker/localsettings.py, line 216 - E265 block comment should start with '# '
* docker/localsettings.py, line 218 - E225 missing whitespace around operator
* docker/localsettings.py, line 219 - E225 missing whitespace around operator
* docker/localsettings.py, line 223 - E265 block comment should start with '# '
* docker/localsettings.py, line 228 - W293 blank line contains whitespace
* docker/localsettings.py, line 231 - W293 blank line contains whitespace
* docker/localsettings.py, line 279 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 280 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 283 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 284 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 294 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 295 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 296 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 298 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 299 - E501 line too long (116 > 115 characters)
* docker/localsettings.py, line 301 - E501 line too long (121 > 115 characters)
* docker/localsettings.py, line 302 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 303 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 307 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 308 - E501 line too long (118 > 115 characters)

**Comment by charlesfleche** [comment]
The commcare image is now build by hub.docker.io when new code is pushed into this branch : https://registry.hub.docker.com/u/charlesfleche/commcare-hq/

It means that it is not required anymore to build the image locally : running docker-run.sh script will download up-to-date version of the CommCareHQ layers.

**Comment by charlesfleche** [comment]
Thanks for mentioning the ansible initiative, I didn't see it at first. The ansible project seems to be more focused on bigger deployment, while with docker I'm aiming at easy laptop / small server distribution. In the short term both projects probably won't overlap. Maybe later we would be able to merge these two. But I'm no devop, so a chat with dimagi's ones would be useful at some point.

**Comment by charlesfleche** [comment]
Thanks for the feedback guys. First let me apologies for the low quality of this PR at first : I was in a rush to push this as I was away from Wednesday evening to Monday and let many small mistakes in the patches. @snopoke , the fixes you've suggested are now applied, thanks.

**Comment by dimagimon** [comment]
There are 67 errors:

* docker/localsettings.py, line 2 - E265 block comment should start with '# '
* docker/localsettings.py, line 2 - W291 trailing whitespace
* docker/localsettings.py, line 15 - E265 block comment should start with '# '
* docker/localsettings.py, line 16 - E501 line too long (127 > 115 characters)
* docker/localsettings.py, line 18 - E265 block comment should start with '# '
* docker/localsettings.py, line 19 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 20 - E262 inline comment should start with '# '
* docker/localsettings.py, line 25 - E265 block comment should start with '# '
* docker/localsettings.py, line 27 - E265 block comment should start with '# '
* docker/localsettings.py, line 30 - E265 block comment should start with '# '
* docker/localsettings.py, line 33 - E265 block comment should start with '# '
* docker/localsettings.py, line 48 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 48 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 52 - E262 inline comment should start with '# '
* docker/localsettings.py, line 52 - E501 line too long (173 > 115 characters)
* docker/localsettings.py, line 54 - E265 block comment should start with '# '
* docker/localsettings.py, line 65 - E261 at least two spaces before inline comment
* docker/localsettings.py, line 68 - E265 block comment should start with '# '
* docker/localsettings.py, line 72 - E265 block comment should start with '# '
* docker/localsettings.py, line 78 - E265 block comment should start with '# '
* docker/localsettings.py, line 89 - E265 block comment should start with '# '
* docker/localsettings.py, line 99 - E202 whitespace before '}'
* docker/localsettings.py, line 101 - E265 block comment should start with '# '
* docker/localsettings.py, line 103 - W291 trailing whitespace
* docker/localsettings.py, line 104 - W291 trailing whitespace
* docker/localsettings.py, line 106 - W291 trailing whitespace
* docker/localsettings.py, line 111 - E265 block comment should start with '# '
* docker/localsettings.py, line 121 - E265 block comment should start with '# '
* docker/localsettings.py, line 160 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 161 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 162 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 163 - E501 line too long (137 > 115 characters)
* docker/localsettings.py, line 164 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 165 - W291 trailing whitespace
* docker/localsettings.py, line 199 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 200 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 208 - E265 block comment should start with '# '
* docker/localsettings.py, line 211 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 211 - E113 unexpected indentation
* docker/localsettings.py, line 212 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 212 - E113 unexpected indentation
* docker/localsettings.py, line 213 - E111 indentation is not a multiple of four
* docker/localsettings.py, line 213 - E113 unexpected indentation
* docker/localsettings.py, line 216 - E265 block comment should start with '# '
* docker/localsettings.py, line 218 - E225 missing whitespace around operator
* docker/localsettings.py, line 219 - E225 missing whitespace around operator
* docker/localsettings.py, line 223 - E265 block comment should start with '# '
* docker/localsettings.py, line 228 - W293 blank line contains whitespace
* docker/localsettings.py, line 231 - W293 blank line contains whitespace
* docker/localsettings.py, line 279 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 280 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 283 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 284 - E122 continuation line missing indentation or outdented
* docker/localsettings.py, line 294 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 295 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 296 - E501 line too long (118 > 115 characters)
* docker/localsettings.py, line 298 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 299 - E501 line too long (116 > 115 characters)
* docker/localsettings.py, line 301 - E501 line too long (121 > 115 characters)
* docker/localsettings.py, line 302 - E501 line too long (117 > 115 characters)
* docker/localsettings.py, line 303 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 307 - E501 line too long (119 > 115 characters)
* docker/localsettings.py, line 308 - E501 line too long (118 > 115 characters)

File: requirements/requirements.txt (hunk 1 above)
Comment (snopoke) [inline requirements/requirements.txt]: @dmyung do you recall what we needed egenix for? Relevant commits below:

ce69b72 [2 years, 4 months ago] re-add egenix-mx-base [Mike White]
9c3eb57 [2 years, 4 months ago] add note about egenix-mx-base to readme [Mike White]
7bf49a9 [2 years, 4 months ago] revert egenix-mx-base requirements.txt addition [Mike White]
2297948 [2 years, 4 months ago] updating requirements - explicit psycopg req, plus egenix-base [dmyung]
57288e4 [2 years, 4 months ago] add egenix-mx-base to requirements.txt [Mike White]

**Comment by snopoke** [comment]
Re lintbot, you can add an ignore line here: https://github.com/dimagi/commcare-hq/blob/master/.lintrc

**Comment by charlesfleche** [comment]
@snopoke I had dangling changes : the merge was a little premature… I was expecting to remove the "Open for review" tag before asking for a merge.

**Comment by NoahCarnahan** [comment]
Hey @charlesfleche, I think you can set which files dimagimon ignores in the [.lintrc](https://github.com/dimagi/commcare-hq/blob/master/.lintrc) file located in the project root. I'm excited to see hq on docker!

**Comment by snopoke** [comment]
Sorry, totally missed that.

**Comment by charlesfleche** [comment]
Can I keep pushing patches to this branch for another merge at some point ? Or should I open a new branch ?

**Comment by snopoke** [comment]
It should be fine to keep pushing to that branch

**Comment by charlesfleche** [comment]
The new patches don't show up in this PR :-/
I'll try to make another PR.

**Pull request merged**